packages/victory-group/src/victory-group.js

```diff
@@ -19,150 +21,209 @@ const fallbackProps = {
   offset: 0
 };
 
-export default class VictoryGroup extends React.Component {
-  static displayName = "VictoryGroup";
-
-  static role = "group";
-
-  static propTypes = {
-    ...CommonProps.baseProps,
-    ...CommonProps.dataProps,
-    children: PropTypes.oneOfType([
-      PropTypes.arrayOf(PropTypes.node),
-      PropTypes.node
-    ]),
-    color: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-    colorScale: PropTypes.oneOfType([
-      PropTypes.arrayOf(PropTypes.string),
-      PropTypes.oneOf([
-        "grayscale",
-        "qualitative",
-        "heatmap",
-        "warm",
-        "cool",
-        "red",
-        "green",
-        "blue"
-      ])
-    ]),
-    horizontal: PropTypes.bool,
-    offset: PropTypes.number
-  };
-
-  static defaultProps = {
-    containerComponent: <VictoryContainer />,
-    groupComponent: <g />,
-    samples: 50,
-    sortOrder: "ascending",
-    standalone: true,
-    theme: VictoryTheme.grayscale
-  };
-
-  static expectedComponents = [
-    "groupComponent",
-    "containerComponent",
-    "labelComponent"
-  ];
-
-  static getChildren = getChildren;
-
-  constructor(props) {
-    super(props);
-    if (props.animate) {
-      this.state = {
-        nodesShouldLoad: false,
-        nodesDoneLoad: false,
-        animating: true
+const usePreviousProps = (props) => {
+  const ref = React.useRef();
+  useEffect(() => {
+    ref.current = props;
+  });
+  return ref.current || {};
+};
+
+const VictoryGroup = (props) => {
+  const { role } = VictoryGroup;
+  const [state, setState] = React.useState({});
+
+  const modifiedProps = Helpers.modifyProps(props, fallbackProps, role);
+  const {
+    eventKey,
+    containerComponent,
+    standalone,
+    groupComponent,
+    externalEventMutations,
+    width,
+    height,
+    theme,
+    polar,
+    horizontal,
+    name
+  } = modifiedProps;
+
+  const childComponents = React.Children.toArray(modifiedProps.children);
+  const calculatedProps = getCalculatedProps(modifiedProps, childComponents);
+  const { domain, scale, style, origin } = calculatedProps;
+
+  // This is a copy of Wrapper.getAnimationProps that doesn't use this.setState
+  const getAnimationProps = React.useCallback(
```

Comment:
We might want to look into refactoring the `Wrapper.getAnimationProps` function so it doesn't call setState inside a getter function, but it would be helpful to have a test case so I know I'm not breaking anything!

Reply:
I don't _think_ the `setState` is actually called from within this getter function. I think it's just used by `Transitions.getTransitionPropsFactory` to set up the functions (including appropriate `setState`) that will be called for each transition. 

Either way, this refactor looks cleaner to me, and I haven't been able to identify anything breaking from the change. If we're moving to using hooks for this, though, it's probably worth also looking at `Wrapper.setAnimationState`